components/test_pd/src/mocker/meta_storage.rs

```diff
@@ -98,14 +111,28 @@ impl PdMocker for MetaStorage {
         ctx.spawn(async move {
             while let Some(x) = watcher.next().await {
                 let mut event = mpb::Event::new();
-                event.set_kv(convert_kv(x.pair));
-                event.set_type(match x.kind {
-                    KvEventType::Put => mpb::EventEventType::Put,
-                    KvEventType::Delete => mpb::EventEventType::Delete,
-                });
+                event.set_kv(convert_kv(x.pair.clone()));
+                match x.kind {
+                    KvEventType::Put => event.set_type(mpb::EventEventType::Put),
+                    KvEventType::Delete => {
+                        event.set_type(mpb::EventEventType::Delete);
+                        event.set_prev_kv(convert_kv(x.pair));
+                    }
+                }
+
                 let mut resp = mpb::WatchResponse::default();
                 resp.set_events(vec![event].into());
                 sink.send((resp, Default::default())).await.unwrap();
+
+                #[cfg(feature = "failpoints")]
```

Comment:
how about move to failpoint test dir?

Reply:
I haven't found a better way to move the failpoint for scenarios that need to be tested in integration test :(